ocean_lib/models/bpool.py

```diff
@@ -24,7 +24,7 @@ class BPool(BToken):
     def __init__(self, *args, **kwargs):
```

Comment:
I propose we take this PR as the opportunity to remove the `BPool._ccontract` instance variable and the `BPool.__init__` method. @calina-c Do you agree?

Reasoning:
The `_ccontract` instance variable is a duplicate. The base class `ContractBase` already has a `contract` instance variable. Anywhere BPool uses `self._ccontract` it should instead use `self.contract.caller`

Reply:
I agree. And it's not a big change :)